.github/workflows/ci.yml

```diff
@@ -0,0 +1,421 @@
+            # use cmake3 instead of default cmake2
+            alternatives --install /usr/local/bin/cmake cmake /usr/bin/cmake3 99
+
+            # To upgrade GCC, one can use GCC 7 from centos-release-scl, required by CGAL.
+            # Even GCC 10+ is provided.
+            # Not used for now, since openEMS is still compatible with GCC 4.8! But we may
+            # need it in the future.
+            # yum install -y devtoolset-7-gcc devtoolset-7-gcc-c++
+            # alternatives --install /usr/local/bin/gcc gcc /opt/rh/devtoolset-7/root/usr/bin/gcc 99
+            # alternatives --install /usr/local/bin/g++ g++ /opt/rh/devtoolset-7/root/usr/bin/g++ 99
+
+            # boost-predef is only available on Boost 1.55 and later.
+            #   /usr/local/include/CGAL/config.h:210:10: fatal error: boost/predef.h:
+            #   No such file or directory
+            # We borrow a copy of Boost 1.58 from rh repo's mariadb backport package.
+            yum install -y rh-mariadb101-boost-devel
+
+            # copy rh repo's Boost to /usr/local because CMake doesn't support CPATH
+            ln -s /opt/rh/rh-mariadb101/root/usr/include/boost /usr/local/include/boost
+            mkdir /usr/local/lib64 && cd /usr/local/lib64
+            for i in /opt/rh/rh-mariadb101/root/usr/lib64/libboost*; do
+                ln -s $i /usr/local/lib64/
+            done
+
+            # not package on CentOS
+            # CGAL v4.14.3 was old enough to be compatible with GCC 4.8. Otherwise,
+            # you get: The compiler feature "cxx_decltype_auto" is not known
+            # to CXX compiler "GNU" version 4.8.5.
+            git clone $GITHUB_SERVER_URL/CGAL/cgal.git --depth=1 --branch=v4.14.3
+            cd cgal && mkdir build && cd build
+            cmake ../ && make && make install
+          else
+            echo "Unknown system!"
+            exit 1
+          fi
+
+      - if: ${{ ! matrix.os.ancient }}
+        name: Checkout CSXCAD.git
+        uses: actions/checkout@v4
+        with:
+          repository: "${{ github.repository_owner }}/CSXCAD"
+          path: CSXCAD
+          # checkout must be deep, not shallow.
+          # We need tags for "git describe", otherwise build fails.
+          fetch-depth: 0
+
+      - if: ${{ ! matrix.os.ancient }}
+        name: Checkout fparser.git
+        uses: actions/checkout@v4
+        with:
+          repository: "${{ github.repository_owner }}/fparser"
+          path: fparser
+
+      - if: ${{ matrix.os.ancient }}
+        name: Checkout CSXCAD.git (legacy system only)
+        run: |
+          git clone $GITHUB_SERVER_URL/$GITHUB_REPOSITORY.git CSXCAD
+          cd $GITHUB_WORKSPACE/CSXCAD
+
+          git fetch --force $GITHUB_SERVER_URL/$GITHUB_REPOSITORY.git "+refs/heads/*:refs/remotes/origin/*"
+          git fetch --force $GITHUB_SERVER_URL/$GITHUB_REPOSITORY.git "+refs/pull/*/merge:refs/remotes/origin/pr/*"
+          git checkout $GITHUB_SHA
```

Comment:
@biergaizi can you explain the reason for this lines? I guess it is to build the correct branch?

I guess the better/correct solution is this line querying the branch directly?:
`git clone -b $GITHUB_REF_NAME $GITHUB_SERVER_URL/$GITHUB_REPOSITORY.git CSXCAD`

Reply:
> I guess the better/correct solution is this line querying the branch directly?: `git clone -b $GITHUB_REF_NAME $GITHUB_SERVER_URL/$GITHUB_REPOSITORY.git CSXCAD`

Probably.